src/app/components/courses-page/courses/courses-item/courses-item.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+import { ICourse } from 'src/app/shared/interfaces/course';
+
+@Component({
+  selector: 'app-courses-item',
+  templateUrl: './courses-item.component.html',
+  styleUrls: ['./courses-item.component.scss'],
+})
+export class CoursesItemComponent {
+  @Input() courseItem: ICourse | undefined = undefined;
```

Comment:
There is no need to initialize variable, you may just leave _@Input() courseItem: ICourse_
If it is undefined )

Reply:
But in that case I get the error:

Property 'courseItem' has no initializer and is not definitely assigned in the constructor.ts(2564)
(property) CoursesItemComponent.courseItem: ICourse.

So, I added "?". Is it correct?

@Input() courseItem?: ICourse;

Reply:
Yes, you are right - the initialization of the Input() was not required before the 16th version of Angular
By adding **?** you highlight that it is not obligatory Input and solve this problem